### PR TITLE
fix(dashmate)!: drive logs configuration

### PR DESCRIPTION
**Pull request by shumkov**
## Issue being fixed or feature implemented

After switching JS Drive to RS Drive the ABCI log file configuration has not been working in dashmate.

## What was done?

- Update RS Drive ABCI log configuration:
   - File path instead of dir path
   - String preset for verbosity levels instead of number
   - CLI verbosity flag changed to 3 items max
   - Define only one default logger (stdout)
   - Hide logs for tests by default
   - Small refactoring
   - Added tests for log rotation
- Set correct paths to Drive ABCI logs for local network

## How Has This Been Tested?

Manually

## Breaking Changes

Drive volume is changed so volumes and containers must be recreated. It means platform file must be wiped.

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [x] I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**
- [x] I have assigned this pull request to a milestone

### CODE REVIEW

File: packages/rs-drive-abci/src/logging.rs [hunk 1]
```diff
@@ -686,34 +763,31 @@ pub fn init_for_tests(verbosity: u8) {
     logger_builder
         .add("default", &config)
         .expect("cannot configure default logger");
+
     logger_builder.build().try_install().ok();
 }
 
 /// Verify log directory path and determine absolute path to log file.
```
Comment (lklimek) [inline packages/rs-drive-abci/src/logging.rs:770]: Please consider the following checks:

```suggestion
/// Verify log file path.
///
/// Ensure that the log file path is correct, that is:
/// - it points to a file, not a directory
/// - if the log file exists, it is writable for the current user
/// - parent directory of the file exists and is writable for current user
/// - path is absolute
```

hint: chatgpt seems to be quite good with tasks like this ;)

File: packages/rs-drive-abci/src/logging.rs [hunk 2]
```diff
@@ -80,6 +152,7 @@ impl Default for LogConfig {
 ///
 /// See https://docs.rs/tracing-subscriber/latest/tracing_subscriber/fmt/index.html#formatters
 #[derive(Debug, Serialize, Deserialize, Default, Clone, Copy)]
+#[serde(rename_all = "camelCase")]
```
[the remark this answers is not in the document]
Comment (shumkov) [reply packages/rs-drive-abci/src/logging.rs:155]: It won't be readable if we have a preset in two words

File: packages/rs-drive-abci/src/logging.rs [hunk 3]
```diff
@@ -686,34 +764,69 @@ pub fn init_for_tests(verbosity: u8) {
...
-            log_dir.to_owned(),
-            "log path must point to an existing directory".to_string(),
+            log_file_path.to_owned(),
+            "log file path must point to file".to_string(),
         ));
     }
-    let path = log_dir.join("drive-abci.log");
-    Ok(path)
+
+    let Some(parent_dir) = log_file_path.parent() else {
```
Comment (lklimek) [inline packages/rs-drive-abci/src/logging.rs:797]: ChatGPT says:

Log file exists but it's read-only: If the log file already exists but it's read-only, the function should return an error. However, this case is not handled by your current implementation.

```
// Assuming /logs/read_only_log.txt is read-only
let result = validate_log_path("/logs/read_only_log.txt");
assert!(result.is_err()); // This assertion will fail with the current implementation
```

File: packages/rs-drive-abci/src/logging.rs [hunk 4]
```diff
@@ -767,15 +768,53 @@ pub fn init_for_tests(level: LogLevelPreset) {
...
+    let log_file_path = log_file_path.as_ref();
+
+    if !log_file_path.is_absolute() {
+        return Err(Error::FilePath(
+            log_file_path.to_owned(),
+            "log file path must be absolute".to_string(),
+        ));
+    }
+
+    if log_file_path.extension().is_none()
```
[the remark this answers is not in the document]
Comment (shumkov) [reply packages/rs-drive-abci/src/logging.rs:788]: Until it exists we can't check. So for now we can just probably have two versions of checks with existing file and not

**Review by lklimek — APPROVED** [review]

**Review by lklimek — DISMISSED** [review]

File: packages/dashmate/test/e2e/testnetFullnode.spec.js [hunk 5]
```diff
@@ -96,6 +96,7 @@ describe('Testnet Fullnode', function main() {
       config = configFile.getConfig(preset);
 
       config.set('dashmate.helper.docker.build.enabled', true);
+      config.set('platform.drive.abci.docker.build.enabled', true);
```
Comment (lklimek) [inline packages/dashmate/test/e2e/testnetFullnode.spec.js:99]: Is this added here by purpose?

File: packages/dashmate/test/e2e/testnetHPFullnode.spec.js [hunk 6]
```diff
@@ -96,6 +96,7 @@ describe('Testnet HP Fullnode', function main() {
       config = configFile.getConfig(preset);
 
       config.set('dashmate.helper.docker.build.enabled', true);
+      config.set('platform.drive.abci.docker.build.enabled', true);
```
Comment (lklimek) [inline packages/dashmate/test/e2e/testnetHPFullnode.spec.js:99]: Is this added here by purpose?

**Review by lklimek — DISMISSED** [review]

File: packages/dashmate/test/e2e/testnetHPFullnode.spec.js (hunk 6 above)
Comment (shumkov) [reply packages/dashmate/test/e2e/testnetHPFullnode.spec.js:99]: Yep

**Pull request merged**